charcoal/compare_taxonomy.py

```diff
@@ -160,6 +160,20 @@ def get_genome_taxonomy(matches_filename, genome_sig_filename, provided_lineage,
     if new_siglist:
         siglist = new_siglist
 
+    # remove duplicate signatures in matches
+    # workaround for issue of duplicate sigs in SBT, see sourmash/#1171
+    new_siglist = []
+    seen_md5 = set()
+    for ss in siglist:
+        ss_md5 = ss.md5sum()
+        if not ss_md5 in seen_md5:
+            new_siglist.append(ss)
+            seen_md5.add(ss_md5)    
+        else:
+            print(f'removing a duplicate match: {ss.name}')
+    if new_siglist:
```

Comment:
curious about this statement - when would there be a non-empty `siglist` and not a non-empty `new_siglist`?

It seems to me like the only time `new_siglist` could be empty is if `siglist` is empty in which case you don't need the `if` statement.

Not a blocker, just curious if this is something you ran into 😓

Reply:
oh, I see. It's based on the code immediately above :)

I think in this case it's appropriate to remove the `if`. I'll do so and merge once the tests pass.